browser/src/layer/tile/ImpressTileLayer.js

```diff
@@ -57,26 +57,33 @@ L.ImpressTileLayer = L.CanvasTileLayer.extend({
 			For a better solution, we need to send the page kinds along with status messages. Then we will check the page kind and set the notes view toggle accordingly.
 		*/
 
-		const isDrawOrNotesPage = ['DrawPage', 'NotesPage'].includes(e.detail.context);
+		const newContext = e.detail.context;
+		const oldContext = e.detail.oldContext;
+		const isDrawOrNotesPage = ['DrawPage', 'NotesPage'].includes(newContext);
 
 		if (isDrawOrNotesPage)
-			app.impress.notesMode = e.detail.context === 'NotesPage';
+			app.impress.notesMode = newContext === 'NotesPage';
 
 		if (app.map.uiManager.getCurrentMode() === 'notebookbar' && isDrawOrNotesPage) {
 			const targetElement = document.getElementById('notesmode');
 			if (!targetElement) return;
 
-			if (e.detail.context === 'NotesPage')
+			if (newContext === 'NotesPage')
 				targetElement.classList.add('selected');
 			else
 				targetElement.classList.remove('selected');
 		}
 
 		if (isDrawOrNotesPage) {
-			this._selectedMode = e.detail.context === 'NotesPage' ? 2 : 0;
+			this._selectedMode = newContext === 'NotesPage' ? 2 : 0;
 			TileManager.refreshTilesInBackground();
 			TileManager.update();
 		}
+
+		if (newContext === 'MasterPage' || oldContext === 'MasterPage') {
```

Comment:
what if both are `MasterPage`? that was most concerning case :)
sorry for that many questions

Reply:
It shouldn't be a problem. Previews are wrong when transition between different page modes happen. But i will check this too. Checking something else now.

No, thanks for the many questions :)